kafka/consumer.go

```diff
@@ -528,8 +529,15 @@ func (c *Consumer) ReadMessage(timeout time.Duration) (*Message, error) {
 // Close Consumer instance.
 // The object is no longer usable after this call.
 func (c *Consumer) Close() (err error) {
-	if !atomic.CompareAndSwapUint32(&c.isClosed, 0, 1) {
-		return getOperationNotAllowedErrorForClosedClient()
+	// Check if the client is already closed.
+	err = c.verifyClient()
+	if err != nil {
+		return err
+	}
+
+	// Client is in the process of closing.
+	if !atomic.CompareAndSwapUint32(&c.isClosing, 0, 1) {
```

Comment:
It is advised to use `UInt32.CompareAndSwap` instead. https://pkg.go.dev/sync/atomic#CompareAndSwapUint32

Though this function is added in Go 1.19. We should figure out a way to make this happen at least in future when min Go is 1.19.

Reply:
Agreed, there are a few more places where we have things to do pending Go version upgrades. I'll note them internally.